frontend/lib/src/components/elements/LinkButton/LinkButton.tsx

```diff
@@ -41,6 +41,8 @@ function LinkButton(props: Readonly<Props>): ReactElement {
   const { colors }: EmotionTheme = useTheme()
   // Material icons need to be larger to render similar size of emojis, emojis need addtl margin
   const isMaterialIcon = element.icon.startsWith(":material")
+  const iconMargin = isMaterialIcon ? "0 sm 0 0" : "0 md 0 0"
```

Comment:
**suggestion (non-blocking):** There are a number of similar lines in each of these components, which presents an opportunity to abstract some of this logic away. WDYT about an approach that encapsulates all of this shared logic into 1 reusable component:

```tsx
export const DynamicButtonIcon: FC<{ icon: string; label?: string }> = ({
  icon,
  label,
}) => {
  const { colors }: EmotionTheme = useTheme()

  if (!icon) {
    return null
  }

  // Material icons need to be larger to render similar size of emojis, emojis need addtl margin
  const isMaterialIcon = icon.startsWith(":material")
  const iconMargin = isMaterialIcon ? "0 sm 0 0" : "0 md 0 0"

  return (
    <DynamicIcon
      size={isMaterialIcon ? "lg" : "base"}
      margin={label ? iconMargin : "0"}
      color={colors.bodyText}
      iconValue={icon}
    />
  )
}
```

Which can then be used by these call sites like:

```tsx
<DynamicButtonIcon icon={element.icon} label={element.label} />
```

The upsides to this approach are that all of the repeated logic can be removed from the call sites, and any fixes to the shared component only need to be applied in 1 place, but will impact all components that use it!

Reply:
Love the idea and really appreciate the feedback! I extended the abstraction a bit, let me know if you think I took it a little too far 😅

Reply:
Very nice! Makes a lot of sense to me given the shared behavior these components have!